source/developers-guide/concept-cart-bundle/index.md

```diff
@@ -0,0 +1,424 @@
+
+namespace CartExtension\Bundle\CartBundle\Domain;
+
+use Doctrine\DBAL\Connection;
+use Shopware\Bundle\CartBundle\Domain\Cart\Cart;
+use Shopware\Bundle\CartBundle\Domain\Cart\CartContextInterface;
+use Shopware\Bundle\CartBundle\Domain\Cart\CartProcessorInterface;
+use Shopware\Bundle\CartBundle\Domain\Cart\ProcessorCart;
+use Shopware\Bundle\CartBundle\Domain\Price\CalculatedDiscount;
+use Shopware\Bundle\CartBundle\Domain\Price\PercentagePriceCalculator;
+
+class NewCustomerDiscountProcessor implements CartProcessorInterface
+{
+    /**
+     * @var Connection
+     */
+    private $connection;
+
+    /**
+     * @var PercentagePriceCalculator
+     */
+    private $percentagePriceCalculator;
+
+    /**
+     * @param Connection $connection
+     * @param PercentagePriceCalculator $percentagePriceCalculator
+     */
+    public function __construct(
+        Connection $connection,
+        PercentagePriceCalculator $percentagePriceCalculator
+    ) {
+        $this->connection = $connection;
+        $this->percentagePriceCalculator = $percentagePriceCalculator;
+    }
+
+    /**
+     * Access point for cart process
+     * @param Cart $cart
+     * @param ProcessorCart $processorCart
+     * @param CartContextInterface $context
+     */
+    public function process(
+        Cart $cart,
+        ProcessorCart $processorCart,
+        CartContextInterface $context
+    ) {
+        //no logged in state?
+        if (!$context->getCustomer()) {
+            return;
+        }
+
+        //validate if customer should get discount
+        if (!$this->isNewCustomer($context->getCustomer()->getId())) {
+            return;
+        }
+
+        //get access to all goods inside the cart
+        $goods = $processorCart->getLineItems()->filterGoods();
+
+        //use core calculator for percentage price calculation with all goods prices
+        $discount = $this->percentagePriceCalculator->calculatePrice(
+            -10,
+            $goods->getPrices(),
+            $context
+        );
+
+        //add calculated discount to cart
+        $processorCart->getLineItems()->add(
+            new CalculatedDiscount('new-customer-discount', $discount)
+        );
+    }
+
+    /**
+     * Validates if the provided customer id is a new customer in the system
+     * and should get the `new customer discount`
+     *
+     * @param int $customerId
+     * @return bool
+     */
+    private function isNewCustomer($customerId)
+    {
+        $hasOrder = $this->connection->createQueryBuilder()
+            ->select('1')
+            ->from('s_order', 'orders')
+            ->where('orders.userID = :id')
+            ->andWhere('orders.ordernumber > 0')
+            ->setMaxResults(1)
+            ->setParameter(':id', $customerId)
+            ->execute()
+            ->fetch(\PDO::FETCH_COLUMN);
+
+        return !$hasOrder;
+    }
+}
+```
+First two conditions validates if a customer is logged in and if the customer has already taken an order.
```

Comment:
The first two conditions validate if a customer is logged in and if the customer has already made an order.